### ISSUE TITLE
mongodb reconnect fails on long running apps

### ISSUE THREAD

**Issue by andre-merzky** [radical-cybertools/radical.pilot#442]
From Dinesh:

```
Exception in thread PilotLauncherWorker-1:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 551, in __bootstrap_inner
    self.run()
  File "/home/dinesh/rp_one/local/lib/python2.7/site-packages/radical.pilot-0.19-py2.7.egg/radical/pilot/controller/pilot_launcher_worker.py", line 209, in run
    limit=BULK_LIMIT
  File "/home/dinesh/rp_one/local/lib/python2.7/site-packages/pymongo-2.7.2-py2.7-linux-i686.egg/pymongo/collection.py", line 1677, in find_and_modify
    **kwargs)
  File "/home/dinesh/rp_one/local/lib/python2.7/site-packages/pymongo-2.7.2-py2.7-linux-i686.egg/pymongo/database.py", line 445, in command
    uuid_subtype, compile_re, **kwargs)[0]
  File "/home/dinesh/rp_one/local/lib/python2.7/site-packages/pymongo-2.7.2-py2.7-linux-i686.egg/pymongo/database.py", line 345, in _command
    for doc in cursor:
  File "/home/dinesh/rp_one/local/lib/python2.7/site-packages/pymongo-2.7.2-py2.7-linux-i686.egg/pymongo/cursor.py", line 1058, in next
    if len(self.__data) or self._refresh():
  File "/home/dinesh/rp_one/local/lib/python2.7/site-packages/pymongo-2.7.2-py2.7-linux-i686.egg/pymongo/cursor.py", line 1002, in _refresh
    self.__uuid_subtype))
  File "/home/dinesh/rp_one/local/lib/python2.7/site-packages/pymongo-2.7.2-py2.7-linux-i686.egg/pymongo/cursor.py", line 915, in __send_message
    res = client._send_message_with_response(message, **kwargs)
  File "/home/dinesh/rp_one/local/lib/python2.7/site-packages/pymongo-2.7.2-py2.7-linux-i686.egg/pymongo/mongo_client.py", line 1210, in _send_message_with_response
    raise AutoReconnect(str(e))
AutoReconnect: [Errno 110] Connection timed out
```

**Comment by marksantcroos** [comment]
Similar?

```
2015:01:16 15:01:09 radical.pilot.MainProcess: [ERROR   ] transfer worker error: [Errno 60] Operation timed out
Traceback (most recent call last):
  File "/Users/mark/.virtualenv/rp/lib/python2.7/site-packages/radical.pilot-0.23-py2.7.egg/radical/pilot/controller/input_file_transfer_worker.py", line 104, in run
    limit=BULK_LIMIT # TODO: bulklimit is probably not the best way to ensure there is just one
  File "build/bdist.macosx-10.9-x86_64/egg/pymongo/collection.py", line 1677, in find_and_modify
    **kwargs)
  File "build/bdist.macosx-10.9-x86_64/egg/pymongo/database.py", line 445, in command
    uuid_subtype, compile_re, **kwargs)[0]
  File "build/bdist.macosx-10.9-x86_64/egg/pymongo/database.py", line 345, in _command
    for doc in cursor:
  File "build/bdist.macosx-10.9-x86_64/egg/pymongo/cursor.py", line 1058, in next
    if len(self.__data) or self._refresh():
  File "build/bdist.macosx-10.9-x86_64/egg/pymongo/cursor.py", line 1002, in _refresh
    self.__uuid_subtype))
  File "build/bdist.macosx-10.9-x86_64/egg/pymongo/cursor.py", line 915, in __send_message
    res = client._send_message_with_response(message, **kwargs)
  File "build/bdist.macosx-10.9-x86_64/egg/pymongo/mongo_client.py", line 1210, in _send_message_with_response
    raise AutoReconnect(str(e))
AutoReconnect: [Errno 60] Operation timed out
2015:01:16 15:01:09 radical.pilot.MainProcess: [DEBUG   ] session 54b9176fa853786677fa
```

**Comment by andre-merzky** [comment]
Yes.  I still have no clue what to do about it...  Probably just reconnect-on-failure...

**Comment by marksantcroos** [comment]
At least on some of the failures indeed.

**Comment by andre-merzky** [comment]
closed in favor of #1129
